### PR TITLE
Upgrade to Spring Cloud Finchley #72

### PR DESCRIPTION
**Pull request by arey**
This pull request upgrades the application to Spring Boot Finchley based on Spring Boot 2.
Local deployment is working. 
Docker deployment is working.

By waiting a patch of Spring Boot 2, I have to add the below line into the application.yml of the  spring-petclinic-microservices-config module:
```
# Temporary hack required by Spring Boot 2
# Waiting issue https://github.com/spring-projects/spring-boot/issues/13042
spring.cloud.refresh.refreshable: false
```

For my point of view, the feature/Finchley branch is complete and could be merge into the master. branch. Are you ok?

### CODE REVIEW

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/OwnerDetails.java [hunk 1]
```diff
@@ -27,19 +28,20 @@
  * @author Maciej Szarlinski
  */
 @Data
+@NoArgsConstructor
```
Comment (mszarlinski) [inline spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/OwnerDetails.java]: Why do we need no-args constructor?

File: spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/OwnerResource.java [hunk 2]
```diff
@@ -74,7 +75,11 @@ public Owner findOwner(@PathVariable("ownerId") int ownerId) {
     @PutMapping(value = "/{ownerId}")
     @Monitored
     public Owner updateOwner(@PathVariable("ownerId") int ownerId, @Valid @RequestBody Owner ownerRequest) {
-        final Owner ownerModel = ownerRepository.findOne(ownerId);
+        final Optional<Owner> owner = ownerRepository.findById(ownerId);
```
Comment (mszarlinski) [inline spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/OwnerResource.java]: `orElseThrow`?

File: spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/PetResource.java [hunk 3]
```diff
@@ -80,7 +86,16 @@ private void save(final Pet pet, final PetRequest petRequest) {
 
     @GetMapping("owners/*/pets/{petId}")
     public PetDetails findPet(@PathVariable("petId") int petId) {
-        return new PetDetails(petRepository.findOne(petId));
+        return new PetDetails(findPetById(petId));
+    }
+
+
+    private Pet findPetById(int petId) {
+        Optional<Pet> pet = petRepository.findById(petId);
```
Comment (mszarlinski) [inline spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/PetResource.java]: `orElseThrow`?

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/OwnerDetails.java (hunk 1 above)
Comment (arey) [reply spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/OwnerDetails.java]: I don't remember why I have to add the @NoArgsConstructor annotation. I will try to remove it.

**Comment by arey** [comment]
Changes applied into the 50231b1 commit

**Comment by arey** [comment]
@mszarlinski may I merge the Fishley branch into master? I've already done a release (and a tag) for the current master version.

**Comment by mszarlinski** [comment]
👍

**Pull request merged**